src/coreclr/src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -81,17 +100,33 @@ private static string MarshalToString(IntPtr arg, string argName)
                     throw new ArgumentNullException(nameof(functionHandle));
                 }
 
-                Delegate d = CreateDelegate(assemblyPath, typeName, methodName, delegateType);
+                IntPtr functionPtr;
+                if (delegateType == null)
+                {
+                    MethodInfo? methodInfo = type.GetMethod(methodName);
```

Comment:
`Delegate.CreateDelegate` binds to both public and non-public methods
`GetMethod(string)` binds to public methods only. I will we should allow non-public methods here too.